### PR TITLE
#1015 add skip option

### PR DESCRIPTION
**Pull request by OlesiaSub**
#1015 
cmd usage: `mvn -Deo.skip=true`
pom.xml usage: add `<skip>true</skip>` to `configuration`

### CODE REVIEW

**Comment by OlesiaSub** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@OlesiaSub how do you know it works? where are the tests?

**Comment by OlesiaSub** [comment]
@yegor256 oh sorry, I wanted to ask you a question about the tests implementation.
I need to check that _nothing will happen_. The only thing I thought of was writing tests similar to your tests for mojos, setting `skip` to `true` there and checking that nothing was created. Is this ok or is there a better way?

**Comment by OlesiaSub** [comment]
You can see an example of such test in the latest commit (it's almost fully copied from PullMojoTest ,so it's a draft)

**Comment by yegor256** [comment]
@OlesiaSub yes, something like this -- run the mojo and assert that nothing is created

**Comment by OlesiaSub** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@mximp please review

File: eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java [hunk 1]
```diff
@@ -121,6 +128,9 @@ abstract class SafeMojo extends AbstractMojo {
     public final void execute() throws MojoFailureException {
         StaticLoggerBinder.getSingleton().setMavenLog(this.getLog());
         try {
+            if (this.skip) {
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java:131]: @OlesiaSub This method now has two `return` points, but should have only 1.

**Review by mximp — CHANGES_REQUESTED** [review]

**Comment by mximp** [comment]
@OlesiaSub please check comments above

File: eo-maven-plugin/src/test/java/org/eolang/maven/SkipTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    @Test
+    public void testExecutedPullMojo(@TempDir final Path temp) {
+        final Path target = temp.resolve("target");
+        this.executePullMojo(temp, target, false);
+        MatcherAssert.assertThat(
+            Files.exists(
+                target.resolve(
+                    String.format(
+                        "%s/org/eolang/io/stdout.eo",
+                        PullMojo.DIR
```
[the remark this answers is not in the document]
Comment (OlesiaSub) [reply eo-maven-plugin/src/test/java/org/eolang/maven/SkipTest.java:54]: For some reason it didn't show any mistakes when I used it locally, but failed multiple times here on pull request. But I made the checks pass here + afaik braces style in PullMojoTest is the same. So could you please clarify why is this braces style wrong?

**Comment by OlesiaSub** [comment]
@mximp fixed, please check

File: eo-maven-plugin/src/test/java/org/eolang/maven/SkipTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mximp) [reply eo-maven-plugin/src/test/java/org/eolang/maven/SkipTest.java:54]: @OlesiaSub According to Checkstyle closing brace should have the same indentation as the line with its opening counterpart (if they're on different lines of course).
I couldn't see the same issue in `PullMojoTests`.

**Review by mximp — APPROVED** [review]

**Comment by mximp** [comment]
@yegor256 please merge

**Comment by yegor256** [comment]
@OlesiaSub usually maven plugins print a message when they are skipped, something like:

```
Execution skipped, due to eo.skip option
```

Would be nice to do the same here

**Comment by OlesiaSub** [comment]
@mximp I added message lo logger.INFO level
(tests are failing again because of #1069 )

**Comment by mximp** [comment]
@Karina5005 Did tests started to fail after your changes? What caused this? It seems that these changes should not affect existing tests.

**Comment by yegor256** [comment]
@mximp that's because `if a { if b {... } }` is a bad style, I believe. should be `if (a & b)`

**Comment by mximp** [comment]
It seems more functional failure:
```
Error:  EOorg.EOeolang.RamTest.writesAndReads(long, int, String, int, int, String)[2]  Time elapsed: 0.338 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: is "hello"
     but: was ""
```

**Comment by Karina5005** [comment]
@mximp It happened after my changes, but the problem is not in my logic.
A test that uses `@Execution(CONCURRENT)` crashes. There was no file junit-platform.properties before my changes, so this test didn't really run in parallel (https://stackoverflow.com/questions/50586201/parallel-test-case-execution-with-junit-5). After my changes, it started to really work in parallel.
I think you should ask @andreoss to look at this test, because his changes added `@Execution(CONCURRENT)` (https://github.com/objectionary/eo/commit/7cca6fbe3320faa1151e7a78836114cfc3cba194) and now it started to work concurrent, but crash .

**Comment by yegor256** [comment]
@OlesiaSub @mximp it's OK to `@Disabled` the test, if it fails for a reason not related to the feature you are implementing. Of course, a ticket must submitted about this situation.

**Comment by yegor256** [comment]
@OlesiaSub see #1069

**Comment by mximp** [comment]
@OlesiaSub would you please comment out the test?

**Comment by OlesiaSub** [comment]
@mximp please help, I added `@Disabled` annotation to the test, but now the tests fail due to style violation
```
Checkstyle: eo-runtime/src/test/java/EOorg/EOeolang/RamTest.java[58]: More than 3 parameters (found 6). (ParameterNumberCheck)
```
Even though I didn't modify anything else... do you have any ideas why this happened?

**Comment by OlesiaSub** [comment]
And I just saw that Yegor experienced the same issue

**Comment by mximp** [comment]
@OlesiaSub did you synched up with master?

**Comment by OlesiaSub** [comment]
@mximp I did when I committed the changes. But I saw that this issue was fixed. Tests pass now.

**Review by mximp — APPROVED** [review]

**Comment by mximp** [comment]
@yegor256 please merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by mximp** [comment]
@OlesiaSub please check

**Comment by mximp** [comment]
@yegor256 can we please try to merge again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**